src/core/default/mappers/simple/bucket_deployer.py

```diff
@@ -69,16 +65,47 @@ def _update_simple_bucket(
 def _remove_simple_bucket(
     transaction_token: str, previous_output: Dict, output_task: OutputTask
 ) -> None:
-
     previous_bucket_name = previous_output.get("bucket_name")
 
+    files = _get_bucket_files(previous_bucket_name)
```

Comment:
Move this to a different function:
```
def _empty_bucket()
```

The idea is to have a very simple function:
```
def _remove_simple_bucket()
    _empty_bucket()
    _delete_empty_bucket()
```
Then inside each function, you update the status:

``` 
def _empty_bucket()
   update progress here
   get files
    iterate over files

def _get_bucket_files()
   update progress here

def _delete_simple_file()
   update progress here


```